enterprise/internal/campaigns/syncer.go

```
@@ -208,6 +211,18 @@ func (s *ChangesetSyncer) SyncChangesetsWithSources(ctx context.Context, bySourc
 	return tx.UpsertChangesetEvents(ctx, events...)
 }
 
+// updateExternalState will update the external state fields on cs based on the current
+// state of the changeset and associated events
+func updateExternalState(cs *campaigns.Changeset, events []*campaigns.ChangesetEvent) {
+	if state, err := cs.State(); err == nil {
```

Shouldn't we log the errors instead of silently dropping them?

---

Done